src/server.c

```diff
@@ -1,9 +1,287 @@
+    if (!(room_handlers)) {
+        error_handler(ERROR_OUT_OF_MEMORY);
+        free(worker_handlers);
+        exit(2);
+    }
+    
+    /**
+     * Inializing a server configuration
+     */
+    struct server_conf *cfg = (struct server_conf*)malloc(sizeof(struct server_conf));
+    if (!(cfg)) {
+        error_handler(ERROR_OUT_OF_MEMORY);
+        free(worker_handlers);
+        free(room_handlers);
+        exit(2);
+    }
+    
+    /**
+     * Make worker pool.
+     */
+    for (int i = 0; i < wnum; i++) {
+        pthread_create(&worker_handlers[i], NULL, &room_fsm, &msgqid);
+    }
+    
+    /**
+     * Make room pool.
+     */
+    for (int i = 0; i < rnum; i++) {
+        pthread_create(&room_handlers[i], NULL, &worker_fsm, &msgqid);
+    }
+    
+    /**
+     * Add a Socket
+     */
+    cfg->socket;
+    if ((cfg->socket = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP)) < 0) {
+        error_handler(ERROR_INIT_SOCKET);
+        free(worker_handlers);
+        free(room_handlers);
+        free(cfg);
+        exit(4);
+    }
+    
+    //TODO: Get right IP adress.
+    struct sockaddr_in addr;
+    memset(&addr, 0, sizeof(struct sockaddr_in));
+    addr.sin_family = AF_INET;
+    addr.sin_addr.s_addr = inet_addr("127.0.0.1");
+    addr.sin_port = htons(PORT);
+    
+    if (bind(cfg->socket, &addr, sizeof(struct sockaddr_in)) < 0) {
+        close(cfg->socket);
+        error_handler(ERROR_BIND_SOCKET);
+        free(worker_handlers);
+        free(room_handlers);
+        free(cfg);
+        exit(5);
+    }
+    
+    /**
+     * Add msgid
+     */
+    cfg->msgid = msgqid;
+    
+    /**
+     * Add pool rooms
+     */
+    cfg->rooms.tid = room_handlers;
+    cfg->rooms.size = rnum;
+    
+    /**
+     * Add pool workers
+     */
+    cfg->workers.tid = worker_handlers;
+    cfg->workers.size = wnum;
+    
+    /**
+     * Add unit point
+     */
+    cfg->units = unit_init(db_path);
+    if (!(cfg->units)) {
+        error_handler(ERROR_INIT_UNITS);
+        free(worker_handlers);
+        free(room_handlers);
+        free(cfg);
+        close(cfg->socket);
+        exit(4);
+    }
+    
+    finalize_conf = cfg;
+    
+    return cfg;
+}
+
+
+/**
+ * Start the receive message loop
+ */
+int loop_recv(int socket, int msgid)
+{
+    struct mbuf msg;
+    int msg_size = sizeof(struct mbuf) - sizeof(long);
+    
+    /**
+     *  Now we are ready to receive messages.
+     */
+    listen(socket, BACKLOG);
+    
+    
+    msg.type = 0; 
+    socklen_t socket_length = sizeof(struct sockaddr_in);
+    struct sockaddr_in client_addr;
+    while (1) {
+        char request[250];
+        
+        int serverd;
+        if ((serverd = accept(socket, &client_addr, &socket_length)) < 0) {
+            error_handler(ERROR_ACCEPT);
+            finalize(finalize_conf);
+        }
+        /**
+         * Get message from client
+         */
+        if (recv(serverd, request, 250, 0) <= 0) {
```

Comment:
You shouldn't receive anything, this is worker's job

Reply:
+